src/components/Quiz/index.tsx

```diff
@@ -0,0 +1,43 @@
+import styled from '@emotion/styled';
+import React, { useState } from 'react';
+import { Quiz as QuizInterface } from '@/interfaces/Quiz';
+import SelectButton from './SelectButton';
+
+const QuizBox = styled.div`
+  border: 1px solid black;
+`;
+
+interface QuizProps {
+  quiz: QuizInterface;
+  index: number;
+  onChangeUserAnswer: (index: number, value: string) => void;
+}
+function Quiz({ quiz, index, onChangeUserAnswer }: QuizProps): JSX.Element {
+  // TODO: OX 버튼 인덱스로 관리할 것
+  const [clickedIndex, setClickedIndex] = useState(-1);
+  const handleClickIndex = (idx: number) => setClickedIndex(idx);
+  return (
+    <QuizBox>
+      <div>Q. {quiz.question}</div>
+      {[
+        ['O', 'true'],
+        ['X', 'false'],
+      ].map(([key, value], idx) => {
+        return (
+          <SelectButton
+            key={key}
+            value={value}
+            onChangeUserAnswer={onChangeUserAnswer}
+            problemIndex={index}
+            currentSelected={clickedIndex}
+            buttonIndex={idx}
+            text={key}
+            onClickCurrentButton={handleClickIndex}
+          />
```

Comment:
흠... 컴포넌트로 만드신 목적이나 활용도에 대해 어떻게 생각하시는지 궁금합니다.
저는 해당 코드를 읽을 때, 편하게 읽는데 어려움을 느꼈고, 그 이유는 컴포넌트 자체가 매우 low한 단계에서 확장성을 크게 가지고 있는 컴포넌트도 아니면서 (즉 quiz로직에 묶여있으면서), 형식은 low한 컴포넌트 인 것과 같이 구성되어있어서라고 생각합니다.
(저의 코드를 읽는 속도나 이해능력에 부족함이라고도 생각합니다 ㅎ..)
컴포넌트화를 할 때 준혁님만의 기준이 있는지 궁금하네요.
제가 구상한다면 어떨까 했을 때, depth를 한 단계 더 안들어가는 것으로 가독성 측면 해결할 수있을 것 같다고 생각했습니다.
이렇게 하면 SelecButton.tsx는 삭제됩니다.
```suggestion
            const SelectButton = styled.button<SelectButtonProps>`
             background-color: ${({ isClicked }) => (isClicked ? 'blue' : 'gray')};
            `;
            <SelectButton
            isClicked={idx === clickedIndex}
            key={key}
            value={value}
            type="button"
            onClick={(e) => {
              onChangeUserAnswer(index, e.currentTarget.value);
              handleClickIndex(idx);
            }}
          >
            {key}
          </SelectButton>
```

Reply:
저도 솔직히 작성하면서 이 부분이 제일 난감했습니다. 앞으로 OX 뿐만 아니라 다양한 방식의 퀴즈 유형에 대응하기 위해서 하위 컴포넌트가 필요하다고 생각했습니다. styled component에 대한 학습이 부족하여 저도 무의식적으로 해당 방식으로 작성한 것 같아요. 가독성 측면은 좀 더 고려해 보겠습니다.